analytic_engine/src/meta/details.rs

```diff
@@ -311,17 +337,26 @@ impl MetaUpdateLogStore for RegionWal {
     }
 
     async fn store(&self, log_entries: &[MetaUpdateLogEntry]) -> Result<()> {
-        let mut log_batch = LogWriteBatch::new(self.region_id);
         let mut payload_batch = Vec::with_capacity(log_entries.len());
 
+        // TODO(ygf11): maybe we can build payload in encode loop.
         for entry in log_entries {
             let payload = MetaUpdatePayload::from(entry);
             payload_batch.push(payload);
         }
 
-        for payload in payload_batch.iter() {
-            log_batch.push(LogWriteEntry { payload });
-        }
+        let region_id = self.region_id;
+        let log_batch_encoder = self
+            .wal_manager
+            .encoder(self.region_id, payload_batch.len() as u64)
+            .await
+            .context(GetLogBatchEncoder {
+                region_id,
+                entries_num: 1u64,
```

Comment:
`entries_num` should be `paylaod_batch.len()`

Reply:
Since we do not allocate SequenceNumber in advance, entries_num is not in the Error now.